src/core.lisp

```diff
@@ -30,23 +30,19 @@
             (project-point-to-camera camera point))
           triangle))
 
+(defun cross-product-2d (a b)
+  (- (* (v:vx a) (v:vy b))
+     (* (v:vy a) (v:vx b))))
+
 (defun triangle-contains (triangle point)
-  (let ((a->p (v:v- point (depth-point-point (first triangle))))
-        (b->p (v:v- point (depth-point-point (second triangle))))
-        (c->p (v:v- point (depth-point-point (third triangle))))
-        (bc-cross (v:vrot2 (v:v- (depth-point-point (second triangle))
-                                 (depth-point-point (third triangle)))
-                           (/ pi 2)))
-        (ca-cross (v:vrot2 (v:v- (depth-point-point (third triangle))
-                                 (depth-point-point (first triangle)))
-                           (/ pi 2)))
-        (ab-cross (v:vrot2 (v:v- (depth-point-point (first triangle))
-                                 (depth-point-point (second triangle)))
-                           (/ pi 2))))
-    (and (< 0 (* (v:v. a->p bc-cross)
-                 (v:v. b->p ca-cross)))
-         (< 0 (* (v:v. b->p ca-cross)
-                 (v:v. c->p ab-cross))))))
+  (destructuring-bind (a b c)
+      (mapcar #'depth-point-point triangle)
+    (let* ((AP (v:vunit* (v:v- point A)))
+           (BP (v:vunit* (v:v- point B)))
+           (CP (v:vunit* (v:v- point C))))
+      (= (float-sign (cross-product-2d BP CP))
```

Comment:
Actually, at this point we calculate barycentric coordinates of the point, which are required to compute its depth. So, in the future we should not discard these values, but reuse them.

Reply:
I'd say we could probably merge `triangle-contains` and `calculate-depth` into one function that returns depth as a second value.